src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadsTest.java

```diff
@@ -1236,5 +1236,91 @@ public void testCisco_CallManager() {
         Assert.assertTrue(report.getSsaEnabled());
     }
 
+    @Test
+    public void testOracleJDK8OnLinux() {
+        testOracleJDK("JAVA_VERSION=\"1.8.11", "Workloads_Oracle_JDK_8_On_Linux", "Oracle JDK 8");
+
+    }
+
+    @Test
+    public void testOracleJDK11OnLinux() {
+        testOracleJDK("JAVA_VERSION=\"11", "Workloads_Oracle_JDK_11_On_Linux", "Oracle JDK 11");
+
+    }
+
+    @Test
+    public void testOracleJDK13OnLinux() {
+        testOracleJDK("JAVA_VERSION=\"13", "Workloads_Oracle_JDK_13_On_Linux", "Oracle JDK 13");
+
+    }
+
+    private void testOracleJDK(String versionCheckText, String ruleNameCheckText, String workloadCheckText)
+    {
+        Map<String, Object> facts = new HashMap<>();
+        // always add a String fact with the name of the agenda group defined in the DRL file (e.g. "SourceCosts")
+        facts.put("agendaGroup", "Workloads");
+
+        VMWorkloadInventoryModel vmWorkloadInventoryModel = new VMWorkloadInventoryModel();
+        Map<String, String> files = new HashMap<>();
+        files.put("/usr/java/latest/release", versionCheckText);
+        vmWorkloadInventoryModel.setFiles(files);
+        List<String> systemServicesNames = new ArrayList<>();
+        systemServicesNames.add("NOTwas");
+        vmWorkloadInventoryModel.setSystemServicesNames(systemServicesNames);
+        facts.put("vmWorkloadInventoryModel", vmWorkloadInventoryModel);
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+
+        Assert.assertEquals(3, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener, "AgendaFocusForTest", "SsaEnabled_System_Services_Present", ruleNameCheckText);
+
+        List<WorkloadInventoryReportModel> reports = Utils.extractModels(GET_OBJECTS_KEY, results, WorkloadInventoryReportModel.class);
+
+        // just one report has to be created
+        Assert.assertEquals(1, reports.size());
+        WorkloadInventoryReportModel report = reports.get(0);
+        Assert.assertNotNull(report.getWorkloads());
+        Assert.assertEquals(1, report.getWorkloads().size());
+        Assert.assertTrue(report.getWorkloads().stream().anyMatch(workload -> workload.toLowerCase().contains(workloadCheckText.toLowerCase())));
+        Assert.assertTrue(report.getSsaEnabled());
+    }
+
+    @Test
+    public void testExcessEscapeCharsNoOracleJDKDetected()
```

Comment:
With my previous comment you could refactor this method and do a call to :
`testOracleJDK("JAVA_VERSION=\"13", null, "Oracle JDK 13");`
Extracting the last lines of the method ( asserts ) to the test body, receiving the `reports` collection as the return of the method.

Reply:
test